.github/workflows/main.yaml

```diff
@@ -33,10 +33,28 @@ jobs:
       workload_identity_provider: ${{ vars.GUNDI_INTEGRATIONS_WORKLOAD_IDENTITY_PROVIDER}}
       service_account: ${{ vars.GUNDI_INTEGRATIONS_SERVICE_ACCOUNT }}
 
+  run_unit_tests:
+    runs-on: ubuntu-latest
+    needs: [vars, build]
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10' 
+      - name: Install pip
+        run: python -m ensurepip --upgrade
+      - name: Install pytest
+        run: pip install pytest
+      - name: Install dependencies
+        run: pip install -r requirements-base.in -r requirements-dev.in
```

Comment:
I think the steps here should be:

- Install pip tools: `pip install pip-tools` 
- Solve dependencies: `pip-compile --output-file=requirements.txt requirements-base.in requirements-dev.in requirements.in` (requirements.in is included because it's used in specific integrations, although it's empty in the template repo)
-  Install dependencies: `pip install -r requirements.txt`

- pytest is in requirements-dev.in so we probably don't need to install it in a separate step